compiler/rustc_const_eval/src/const_eval/machine.rs

```diff
@@ -252,6 +252,7 @@ impl<'tcx> CompileTimeEvalContext<'tcx> {
                 ty::ParamEnv::reveal_all(),
                 const_def_id,
                 instance.args,
+                Some(self.find_closest_untracked_caller_location()),
```

Comment:
Why does resolving now need the caller_location...?

If it needs the call site span, it should use `self.cur_span()`.

Reply:
> Why does resolving now need the caller_location...?

So we can provide a relevant span if we hit type overflow.

> If it needs the call site span, it should use self.cur_span().

Thanks. There's basically no documentation for when to use either, other than "This is the fancy version of `cur_span`", lol.

Reply:
find_closest_untracked_caller_location is specifically honoring `#[caller_location]` to show a span in the caller instead, walking up the call stack.